src/app/directives/datatable/pagination.directive.ts

```diff
@@ -0,0 +1,279 @@
+ *        {{ value }}
+ *      </ng-template>
+ *    </ngx-datatable-column>
+ * </ngx-datatable>
+ * ```
+ *
+ * ## Example with associated models
+ *
+ * We can't directly use the value, as angular will not detect changes unless
+ * the user hovers over the row, leaving the loading animation forever until
+ * then.
+ *
+ * ```html
+ * <ngx-datatable
+ *  bawDatatableDefaults
+ *  [bawDatatablePagination]="{ filters: filters, getModels: getModels}">
+ *   <ng-template let-row="row" ngx-datatable-cell-template>
+ *     <baw-loading
+ *       *ngIf="row.site | isUnresolved; else site"
+ *       size="sm"
+ *     ></baw-loading>
+ *     <ng-template #site>
+ *       <a [bawUrl]="row.site.viewUrl">
+ *         {{ row.site.name }}
+ *       </a>
+ *     </ng-template>
+ *   </ng-template>
+ * </ngx-datatable>
+ * ```
+ */
+@Directive({
+  selector: "[bawDatatablePagination]",
+})
+export class DatatablePaginationDirective<Model extends AbstractModel>
+  extends withUnsubscribe()
+  implements AfterContentInit
+{
+  /**
+   * @param filters Base api filters for table. If this is an observable, on
+   * trigger, it will update the table to match the new filters. If this is a
+   * plain object, it will be used directly. This value is not compatible with
+   * the async pipe
+   * @param getModels Get models for the table. This will be fed the filters
+   * for the current page of the table, and should output the matching models
+   */
+  // eslint-disable-next-line @typescript-eslint/quotes
+  @Input("bawDatatablePagination") public pagination: {
+    filters: BehaviorSubject<Filters<Model>> | Filters<Model>;
+    getModels: (filters: Filters<Model>) => Observable<Model[]>;
+  };
+
+  /** Base API filters, this is extracted from the pagination input */
+  private filters$: Observable<Filters<Model>>;
+  /**
+   * Observable which outputs each table row. Each row value will be the model
+   * retrieved from the api request
+   */
+  private rows$: Observable<any[]>;
+  /**
+   * Observable tracking the total number of models for the current base filter
+   */
+  private total$: Observable<number>;
+  /** Observable tracking when the table is loading new data */
+  private loading$ = new BehaviorSubject<boolean>(false);
+  /**
+   * Observable tracking any changes to the current page, or sort, of the table
+   */
+  private pageAndSort$ = new BehaviorSubject<PageAndSort<Model>>({ page: 0 });
+
+  public constructor(@Host() private datatable: DatatableComponent) {
+    super();
+  }
+
+  public ngAfterContentInit(): void {
+    // Convert basic filters to observable
+    this.filters$ =
+      this.pagination.filters instanceof BehaviorSubject
+        ? this.pagination.filters
+        : of(this.pagination.filters);
+
+    this.filters$.pipe(takeUntil(this.unsubscribe)).subscribe(() => {
+      this.pageAndSort$.next({
+        page: 0,
+        sort: this.pageAndSort$.getValue().sort,
+      });
+    });
```

Comment:
No docs for this. I think this is resetting the page and (not?) the sort when the filter changes?